mediator-coordination/src/plugin.rs

```diff
@@ -33,12 +88,46 @@ impl Plugin for MediatorCoordinationPlugin {
 
     fn routes(&self) -> Router {
         let msg = "This should not occur following successful mounting.";
-        let storage_dirpath = std::env::var("STORAGE_DIRPATH").expect(msg);
+        let MediatorCoordinationPluginEnv {
+            public_domain,
+            storage_dirpath,
+            mongo_uri,
+            mongo_dbn,
+        } = load_plugin_env().expect(msg);
 
+        // Load crypto identity
         let mut fs = StdFileSystem;
         let diddoc = util::read_diddoc(&fs, &storage_dirpath).expect(msg);
         let keystore = util::read_keystore(&mut fs, &storage_dirpath).expect(msg);
 
-        web::routes(diddoc, keystore)
+        // Load connection to database
+        let db = load_mongo_connector(&mongo_uri, &mongo_dbn).expect(msg);
```

Comment:
This can also be saved during mounting, and reused here.

Reply:
:+1:

Reply:
This does not appear to be addressed.

Reply:
can't find a better way to refactor this without creating global variables which may create some overheads